cpp/include/raft/spatial/knn/detail/ivf_pq_search.cuh

```diff
@@ -42,6 +42,19 @@
 
 #include <cuda_fp16.h>
 
+// Disable warnings like 'error: variable "base_diff" was declared but never referenced'
+// and 'error: missing return statement at end of non-void function'
+#define RAFT_PRAGMA(X) _Pragma(#X)
```

Comment:
I generally avoid supressing things like this because it undermines the warnings as errors flags that we've intentionally set on the built to fix these types of things. 

Tagging @robertmaynard for any potential ideas here. Perhaps if we can't figure out another way around this, and it really is happening because of the gcc version, we should consider supressing conditionally depending on the GCC version.